server/server.js

```diff
@@ -294,25 +297,26 @@ app.get('/logout', (req, res) => {
     }
 });
 
-app.post('/flic_button_press', Validator.header(['button-serial-number','button-battery-level']).exists(), async (req, res) => {
+app.post('/flic_button_press', Validator.header(['button-serial-number']).exists(), async (req, res) => {
 
     try {
         const validationErrors = Validator.validationResult(req)
 
         if(validationErrors.isEmpty()){
-            let button = await db.getButtonWithSerialNumber(req.get('button-serial-number'))
+            const serialNumber = req.get('button-serial-number')
+            const batteryLevel = req.get('button-battery-level')
+
+            let button = await db.getButtonWithSerialNumber(serialNumber)
             if(button === null) {
-                log(`Bad request: Serial Number is not registered. Serial Number is ${req.get('button-serial-number')}`)
+                log(`Bad request: Serial Number is not registered. Serial Number is ${serialNumber}`)
                 res.status(400).send();
-            }
-            else {
-                let numPresses = 1
+            } else {
+                await handleValidRequest(button, 1, batteryLevel)
 
-                if(req.query.presses == 2) {
-                    numPresses = 2;
+                if (req.query.presses == 2) {
+                    await handleValidRequest(button, 1)
```

Comment:
@schwarrrtz @mariocimet 

This change is the only really important one. I think that we should get this deployed sooner rather than later because without it, we end up the situation where if the resident's first press is considered "numPresses = 2", then we don't send an initial message. It's only an issues at the Princess Rooms.